### PR TITLE
Refactor the Hunger Meter, add Machine People Battery Meters

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
Refactors the hunger meter so it can support `species` level hunger variables, and cut down on the possible switching statements. The thrown alert also supports icon overriding as well now. In the case of hunger processing, it bases itself off the `species`' `hunger_icon` and `hunger_type`.

Hunger sprites are also their own separate files, with shared sprite names.

Adds distinct "hunger" meters for Machine People, with their own icons, names and descriptions.
I'm open to suggestions in general.
- "Full Charge" - "Your cell is at full charge. Might want to give APCs some space."
- "High Charge" - "You're almost all charged, but could top up a bit more."
- "Half Charge" - "You feel moderately charged, but a bit more juice couldn't hurt."
- "Low Charge" - "Could use a little charging right about now."
- "Nearly Discharged" - "You're almost drained. The low power makes moving around a chore."

## Why It's Good For The Game
The burger is odd, and so are the descriptions. Less switch, just set the icon and type in species.

## Images of changes
IPC_Battery
Battery2

https://user-images.githubusercontent.com/80771500/162831583-3b9c52ee-1b15-487b-b5ad-de22b8944159.mp4

## Changelog
:cl:
add: Machine People battery meter
/:cl:

### CODE REVIEW

File: code/_onclick/hud/alert.dm [hunk 1]
```diff
@@ -184,6 +184,34 @@
...
+	icon_state = "fed_machine"
+
+/obj/screen/alert/hungry/machine
+	name = "Low Charge"
+	desc = "Could use a little charging right about now."
+	icon_state = "hungry_machine"
+
+/obj/screen/alert/starving/machine
+	name = "Nearly Discharged"
+	desc = "You're almost drained. The low power make moving around a chore."
```
Comment (warriorstar-orion) [inline code/_onclick/hud/alert.dm:211]:
```suggestion
	desc = "You're almost drained. The low power makes moving around a chore."
```

**Comment by Miraviel** [comment]
This is really cool!

May I ask for a little sprite change, though?

Instead of an empty cell:

image

I suggest adding the green light to it, which indicates the cell is charged!

image

You can do so by getting the cell from `icons\obj\power.dmi` named `hcell` (or its variants) and adding the `cell-o2` overlay to it from the same file!

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -790,35 +790,25 @@
...
+			new_hunger = "well_fed"
+		if(NUTRITION_LEVEL_HUNGRY to NUTRITION_LEVEL_FED)
+			new_hunger = "fed"
+		if(NUTRITION_LEVEL_STARVING to NUTRITION_LEVEL_HUNGRY)
+			new_hunger = "hungry"
+		else
+			new_hunger = "starving"
+	if(dna.species.hunger_type)
+		new_hunger += "/[dna.species.hunger_type]"
+	if(!(dna.species.hunger_level == new_hunger))
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/human/life.dm:809]:
```suggestion
	if(dna.species.hunger_level != new_hunger)
```

**Review by hal9000PR — APPROVED** [review]
lgtm!

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**